a3_Arrays_Methods/Program.cs

```diff
@@ -0,0 +1,169 @@
+        {
+            //Задача:
+            //Найти наибольший элемент матрицы A(10, 10) и записать ноль в ту ячейку,
+            //где он находится. Вывести наибольший элемент, исходную и полученную матрицу.
+            //Массив под измененную версию не нужен.
+
+            int[,] Array = new int[10, 10];
+            int MaxElement = 0;
+
+            for (int i = 0; i < Array.GetLength(0); i++)
+            {
+                for (int j = 0; j < Array.GetLength(1); j++)
+                {
+                    Random rand = new Random();
+                    Array[i, j] = rand.Next(100, 1000);
+                    Console.Write(Array[i, j] + " ");
+
+                    if (MaxElement < Array[i, j])
+                    {
+                        MaxElement = Array[i, j];
+                    }
+                }
+                Console.Write("\n");
+            }
+
+            Console.WriteLine("\nNew array.");
+            for (int i = 0; i < Array.GetLength(0); i++)
+            {
+<<<<<<< Updated upstream
+                for (int j = 0;  j < Array.GetLength(1);  j++)
+=======
+                for (int j = 0; j < Array.GetLength(1); j++)
+>>>>>>> Stashed changes
+                {
+                    if (Array[i, j] != MaxElement)
+                    {
+                        Console.Write(Array[i, j] + " ");
+                    }
+                    else if (Array[i, j] == MaxElement)
+                    {
+                        Array[i, j] = 0;
+                        Console.Write(Array[i, j] + " ");
+                    }
+                }
+                Console.Write("\n");
+            }
+            Console.WriteLine("\n" + MaxElement + " max number.");
+        }
+
+        static void GetLocalMaximum()
+        {
+            //Дан массив целых чисел из 30 элементов.
+            //Найдите все локальные максимумы и вывести их.
+            //(Элемент является локальным максимумом, если он не имеет соседей, больших, чем он сам)
+            //Крайние элементы являются локальными максимумами если не имеют соседа большего, чем они сами.
+            //Программа должна работать с массивом любого размера.
+            //Массив всех локальных максимумов не нужен.
+
+            int[] Array = new int[30];
+            Random rand = new Random();
+
+            for (int i = 0; i < Array.Length; i++)
+            {
+                Array[i] = rand.Next(0, 10);
+                Console.Write(Array[i] + " ");
+            }
+
+            Console.WriteLine("\n");
+            for (int i = 0; i < Array.Length; i++)
+            {
+                //Console.WriteLine(Array.Length);
+                if (Array.Length > 1)
+                {
+                    if (i == 0 && Array[i] > Array[(i + 1)])
+                    {
+                        Console.Write(Array[i] + " ");
+                    }
+                    else if (Array.Length != (i + 1) && i > 0 && Array[i] > Array[(i - 1)] && Array[i] > Array[(i + 1)])
```

Comment:
To much comparisons inside if statement. It's better to split to different methods or simplify them. Please review an updated example below .